DataFormats/Headers/include/Headers/DataHeader.h

```diff
@@ -489,11 +489,25 @@ auto get(const o2::byte* buffer, size_t /*len*/ = 0)
   const BaseHeader* current = BaseHeader::get(buffer);
   if (!current)
     return HeaderConstPtrType{nullptr};
-  if (current->description == HeaderValueType::sHeaderType)
+  if (current->description == HeaderValueType::sHeaderType) {
+    // FIXME: for the moment we require the version to exactly match, but we need to think how
+    // to handle multiple versions
+    if (current->headerSize != sizeof(HeaderValueType) || current->sVersion != HeaderValueType::sVersion) {
+      throw std::runtime_error("invalid header version");
+    }
     return reinterpret_cast<HeaderConstPtrType>(current);
-  while ((current = current->next())) {
-    if (current->description == HeaderValueType::sHeaderType)
+  }
+  while (current->flagsNextHeader) {
+    current = current->next();
+    if (!current) {
+      throw std::runtime_error("inconsistent header stack, no O2 header at expected offset");
+    }
+    if (current->description == HeaderValueType::sHeaderType) {
+      if (current->headerSize != sizeof(HeaderValueType) || current->sVersion != HeaderValueType::sVersion) {
+        throw std::runtime_error("invalid header version");
+      }
       return reinterpret_cast<HeaderConstPtrType>(current);
+    }
```

Comment:
Could you move it to the .cc file? Throwing exceptions inline might explode the code size as stack unwinding might be embedded in every call site.

Reply:
yes, I thought the same, also to avoid having the same code block twice